src/transforms/operationTransforms.ts

```diff
@@ -104,9 +104,10 @@ export function transformOperationSpec(
 
   const hasMultipleRequests = operationDetails.requests.length > 1;
   for (const request of operationDetails.requests) {
-    const isXML = request.mediaType
-      ? request.mediaType === KnownMediaType.Xml
-      : operationDetails.mediaTypes.has(KnownMediaType.Xml);
+    const isXML =
```

Comment:
@joheredi There is a situation where the operationDetails has 1 mediatype of 'xml'/2 mediatypes of binary & xml, but the request mediatype is set to the first option of binary. In such a case, instead of considering xml in the mediatypes, the above code assumes isXML to false. 

My code change will fix it.